wporg-gp-translation-events.php

```diff
@@ -162,12 +162,58 @@ public function event_meta_boxes() {
 	 */
 	public function event_dates_meta_box( WP_Post $post ) {
 		wp_nonce_field( 'event_dates_nonce', 'event_dates_nonce' );
-		$event_start = get_post_meta( $post->ID, '_event_start', true );
-		$event_end   = get_post_meta( $post->ID, '_event_end', true );
-		echo '<label for="event_start">Start Date: </label>';
-		echo '<input type="date" id="event_start" name="event_start" value="' . esc_attr( $event_start ) . '" required>';
-		echo '<label for="event_end">End Date: </label>';
-		echo '<input type="date" id="event_end" name="event_end" value="' . esc_attr( $event_end ) . '" required>';
+		$event_start    = get_post_meta( $post->ID, '_event_start', true );
+		$event_end      = get_post_meta( $post->ID, '_event_end', true );
+		$event_timezone = get_post_meta( $post->ID, '_event_timezone', true );
```

Comment:
I think it would make sense to use `self::get_event_repository()->get_event( $post->ID )` here, which will return an instance of `Event`. Then in the code below we can use the getters on the event instance to access the event properties, e.g. `$event->start()`.

Reply:
The reason I went for using the post_meta directly is that I see this as a way to show the "real" database contents to the admin. Let's say we have a bug in our date functions, then here we can verify (and potentially fix) what's in the database since we show and save the data unmodified by our logic.

Reply:
I see. In that case, since the purpose is to document for an admin how the data is stored, we could instead add that information as a comment. The purpose of the repositories was to abstract how the data is stored from the rest of the codebase, so it defeats the purpose a bit to not use them here. I understand your point, but personally I would optimise for making the code easier to maintain, instead of optimising for a situation that I think might not happen very often.

Just my perspective, I'm fine with leaving it as is. Though I think we shouldn't extend this practice to other places, if we need to do more things in wp-admin.

Reply:
I think I tend to agree after all, so I have made your suggested changes.